.github/workflows/snyk.yml

```diff
@@ -0,0 +1,42 @@
+name: CI / snyk.io
+
+on:
+  push:
+    branches:
+      - "**"
+      - "!main"
+
+jobs:
+  snyk:
+    runs-on: [ubuntu-latest]
+    strategy:
+      matrix:
+        target:
+          [
+            dependency-test,
+            code-test,
+          ]
+        include:
+          - target: dependency-test
+            name: Snyk dependency scan
+            cmd: snyk test --all-projects --exclude=misc --severity-threshold=high
+            continue-on-error: true
+          - target: code-test
+            name: Snyk code analysis
+            cmd: snyk code test --severity-threshold=high
+            continue-on-error: false
+    name: "${{ matrix.name }}"
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - uses: snyk/actions/setup@master
+      - uses: actions/setup-go@v1
+        with:
+          go-version: '1.18'
```

Comment:
Just noticed but probably applies to all this snyk prs, why 1.18 and not more recent one?

Reply:
As I mentioned to Diego in the chat, these were mostly copy & paste from a configuration that Brian made.

Reply:
I used 1.18 on the original PR I opened for object-storage-service, since that was the version listed in the `go.mod` file.